tests/qmctl-test/scripts/qmctl_json_output.sh

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+. ../e2e/lib/utils
+
+QMCTL_SCRIPT="../../tools/qmctl/qmctl"
+TEST_RESULTS=0
+
+info_message "Starting qmctl JSON output tests..."
+
+# Test 1: Show commands with JSON output
+run_test "JSON Show container" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" show container --json
+
+# Handle unix-domain-sockets separately (may have missing ss command)
+info_message "Testing: JSON Show unix-domain-sockets"
+unix_json_file=$(mktemp)
+if python3 "$QMCTL_SCRIPT" show unix-domain-sockets --json > "$unix_json_file" 2>&1; then
+    unix_json_exit=0
+else
+    unix_json_exit=$?
+fi
+
+if [ "$unix_json_exit" -eq 0 ]; then
+    if python3 -m json.tool < "$unix_json_file" > /dev/null 2>&1; then
+        pass_message "JSON Show unix-domain-sockets - Valid JSON output"
+    else
+        fail_message "JSON Show unix-domain-sockets - Invalid JSON output"
+        TEST_RESULTS=1
+    fi
+elif [ "$unix_json_exit" -eq 1 ] && grep -q "ss.*not found\|executable file.*ss.*not found" "$unix_json_file"; then
+    if python3 -m json.tool < "$unix_json_file" > /dev/null 2>&1; then
+        info_message "SKIP: JSON Show unix-domain-sockets - Missing 'ss' command (valid JSON error)"
+    else
+        fail_message "JSON Show unix-domain-sockets - Invalid JSON error response"
+        TEST_RESULTS=1
+    fi
+else
+    fail_message "JSON Show unix-domain-sockets - Unexpected error (exit $unix_json_exit)"
+    TEST_RESULTS=1
+fi
+rm -f "$unix_json_file"
+
+run_test "JSON Show shared-memory" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" show shared-memory --json
+
+run_test "JSON Show available-devices" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" show available-devices --json
+
+run_test "JSON Show namespaces" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" show namespaces --json
+
+# Test 2: Exec commands with JSON output
+info_message "QM container is available - testing exec JSON commands"
+
+run_test "JSON Exec echo" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" exec --json echo "hello world"
+
+# Handle hostname separately (may not work in container)
+info_message "Testing: JSON Exec hostname"
+hostname_json_file=$(mktemp)
+if python3 "$QMCTL_SCRIPT" exec --json hostname > "$hostname_json_file" 2>&1; then
+    hostname_json_exit=0
```

Comment:
do we need the `hostname_json_exit=0` ? Can we do directly do `hostname_json_exit=$?` and avoid if/else ?

Reply:
python3 "$QMCTL_SCRIPT" exec --json hostname > "$hostname_json_file" 2>&1
hostname_json_exit =  $?

isn't the same? you get the output err code so you don't need to do if to get the value.

Reply:
It seems the same, but it doesn't work because` python3 "$QMCTL_SCRIPT" exec --json hostname  `is failing, and we need to catch this failure.
So there are two options 
1. Use if-else
2.U se set +e